opencog/reasoning/pln/rules/formulas.scm

```diff
@@ -0,0 +1,78 @@
+; =============================================================================
+; A list of the helper functions necessary in different inference rules
+; -----------------------------------------------------------------------------
+; Index
+; - inversion-formula
+; - simple-deduction-formula
+; - find-replace
+;------------------------------------------------------------------------------
+
+
+; =============================================================================
+; Inversion Formula
+; sBA = (sAB * sB)/sA
+; -----------------------------------------------------------------------------
+
+(define (inversion-formula sAB sA sB)
+	(/ (* sAB sB) sA))
+
+; =============================================================================
+; Simple Deduction Formula
+; IF: 0 <= sAB-max((sA+sB-1)/sA, 0) 0<= min(1, sB/sA) -max((sA+sB-1)/sA, 0)
+; AND 0 <= sBC-max((sB+sC-1)/sB, 0) 0<= min(1, sC/sB)-max((sB+sC-1)/sB, 0)
+; sAC = [sAB.sBC + ((1-sAB)(sC - sBsBC))/(1-sB)] 
+; ELSE:
+; sAC = 0
+; -----------------------------------------------------------------------------
+
+; Consistency Conditions
+(define (consistency1 sA sB sAB)
+	(- sAB
+		(max 
+			(/ 
+				(- 
+					(+ sA sB) 
+					1) 
+				sA) 
+			0)))
+
+(define (consistency2 sA sB sAB)
+	(- 
+		(min 
+			1 
+			(/ sB sA)) 
+		sAB))
+
+; Main Formula
+
+(define (simple-deduction-formula sA sB sC sAB sBC)
+	(if
+		(and
```

Comment:
1. on the consistency formulas sA=0 and sB=0 are not considered.
2. the relations between the consistency checks isn't correct.
3. https://github.com/opencog/opencog/pull/1593/files#diff-087a7197c73c81976487cb366eedb369L130 is missing.